Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -449,6 +444,21 @@ private LanguageServerSettings.PythonSettings GetSettings(Uri scopeUri = null)
             return settings;
 
         }
+        private void OnInterpreterChanged(object sender, EventArgs e) {
+
+            // By default Pylance will tell us to watch everything under the workspace with pattern "**/*"
+            // we can exclude the interpreter directory because we already have package managers listening to them
+            this._clientContexts.ForEach(context => {
+
+                if (PathUtils.IsSubpathOf(context.RootPath, context.InterpreterConfiguration.InterpreterPath)) {
+                    var pattern = CommonUtils.GetRelativeFilePath(context.RootPath, context.InterpreterConfiguration.GetPrefixPath()) + "/**/*";
+                    var watcher = new FileSystemWatcher() { GlobPattern = pattern };
+                    this._fileListener.AddExclude(watcher);
```

Comment:
this is the fix. just exclude files under the interpreter if the interpreter is under the root workspace